File: PolyclinicsSystemBackend/PolyclinicsSystemBackend/Data/Entities/Appointment/Appointment.cs
using System;
using System.Collections.Generic;
using PolyclinicsSystemBackend.Data.Entities.MedicalCard;
using PolyclinicsSystemBackend.Dtos.Appointment;
using PolyclinicsSystemBackend.Dtos.Chat;

namespace PolyclinicsSystemBackend.Data.Entities.Appointment
{
    public class Appointment
    {
        public int AppointmentId { get; set; }

        public DateTime AppointmentDate { get; set; }

        public AppointmentStatuses AppointmentStatus { get; set; }

        public string DoctorId { get; set; }

        public User Doctor { get; set; }

        public string PatientId { get; set; }

        public User Patient { get; set; }

        public int? DiagnoseId { get; set; }

        public Diagnose? Diagnose { get; set; }

        public List<Message> Messages { get; set; }
    }
}

File: PolyclinicsFrontEnd/src/app/shared/models/appointment/appointment.ts
import {AppointmentStatuses} from "../../enums/appointment-statuses";
import {Doctor} from "../doctor/doctor";
import {Patient} from "../patient/patient";
import {Diagnose} from "../diagnose/diagnose";

export class Appointment {
  public appointmentId: number = -1
  public appointmentStatus: AppointmentStatuses = AppointmentStatuses.NotExist
  public doctor: Doctor = new Doctor()
  public patient: Patient = new Patient()
  public diagnose?: Diagnose = new Diagnose()
  public appointmentDate: string = ''
}
